File: test cases/frameworks/17 mpi/main.f90
program main
use, intrinsic :: iso_fortran_env, only: stderr=>error_unit
use mpi

implicit none

logical :: flag
integer :: ier

call MPI_Init(ier)

if (ier /= 0) then
  write(stderr,*) 'Unable to initialize MPI', ier
  error stop
endif

call MPI_Initialized(flag, ier)
if (ier /= 0) then
  write(stderr,*) 'Unable to check MPI initialization state: ', ier
  error stop
endif

if (.not.flag) error stop "MPI did not initialize!"

call MPI_Finalize(ier)
if (ier /= 0) then
  write(stderr,*) 'Unable to finalize MPI: ', ier
  error stop
endif

print *, "OK: Fortran MPI"

end program

File: test cases/frameworks/17 mpi/main.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
    int ier, flag, L;
    char version[MPI_MAX_LIBRARY_VERSION_STRING];

    MPI_Get_library_version(version, &L);
    printf("%s", version);

    ier = MPI_Init(&argc, &argv);
    if (ier) {
        fprintf(stderr, "Unable to initialize MPI: %d\n", ier);
        return 1;
    }
    ier = MPI_Initialized(&flag);
    if (ier) {
        fprintf(stderr, "Unable to check MPI initialization state: %d\n", ier);
        return 1;
    }
    if (!flag) {
        fprintf(stderr, "MPI did not initialize!\n");
        return 1;
    }
    ier = MPI_Finalize();
    if (ier) {
        fprintf(stderr, "Unable to finalize MPI: %d\n", ier);
        return 1;
    }
    return 0;
}
